### PR TITLE
Ticket 3929 - MOXA e1210: Added test that checks the counter correctly increases.

### PR DESCRIPTION
**Pull request by aaron-long**
Each DI can be triggered which will increament the DI's associated counter (EPICS layer record). Typically we want these counter values to remain persistent even on restart (as they are a safety feature). 

The new tests trigger each DI and then check their counter has incremented correctly. The resetDICounter function allows for these to be reset, but add considerable time to the test length, as such it shouldn't be in setUP.

### CODE REVIEW

File: tests/moxa1210.py [hunk 1]
```diff
@@ -55,3 +66,12 @@ def test_WHEN_DI_input_is_switched_on_THEN_only_that_channel_readback_changes_to
                 continue
 
             self.ca.assert_that_pv_is("CH{:02d}:DI".format(test_channel), "Low")
+
+    @parameterized.expand([
+        ("CH{:02d}:DI:CNT".format(channel), channel) for channel in CHANNELS
+    ])
+    def test_WHEN_di_input_is_triggered_THEN_di_counter_increases(self, channel_pv, channel):
+        self.resetDICounters()
```
Comment (Alistair-McGann-Tessella) [inline tests/moxa1210.py]: This function appears to reset all of the counters when testing one channel. Might it speed the running of the tests up by only resetting each channel as it's being tested?

File: tests/moxa1210.py [hunk 2]
```diff
@@ -55,3 +66,12 @@ def test_WHEN_DI_input_is_switched_on_THEN_only_that_channel_readback_changes_to
...
+        self._lewis.backdoor_run_function_on_device("set_di", (channel, (True,)))
+
+        self.ca.assert_that_pv_is(channel_pv, 1)
```
Comment (Alistair-McGann-Tessella) [inline tests/moxa1210.py]: Would it be useful to demonstrate that the counters are properly adding up over multiple switches by turning a DI on and off again a few times?

File: tests/moxa1210.py (hunk 1 above)
Comment (aaron-long) [reply tests/moxa1210.py]: I changed the reset function to update only the channel the test applies to.

File: tests/moxa1210.py (hunk 2 above)
Comment (aaron-long) [reply tests/moxa1210.py]: Created a new test to check that individual channels would correctly log multiple events.

**Pull request merged**